webconsole/src/main/java/org/apache/felix/webconsole/internal/core/BundlesServlet.java

```diff
@@ -1710,6 +1718,43 @@ private String getSymbolicName( File bundleFile )
         // fall back to "not found"
         return null;
     }
+    
+    private String getBundleVersion( File bundleFile )
+    {
+        JarFile jar = null;
+        try
+        {
+            jar = new JarFile( bundleFile );
```

Comment:
try with resources (https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)?

Reply:
The Felix Webconsole project is currently marked to be Java 6 compliant - this was just reproducing a pattern used already before for bsn - I just changed this based on suggestion by Carsten below to make sure to use a single method to obtain bsn & version